data/tutorials/language/0lg_09_env.md

```diff
@@ -0,0 +1,675 @@
+
+Since, by definition, effects lie outside function types, a function type can't reflect the effects a function may have. However, it is important to document the intended side effects a function may have. Consider the `Unix.time` function. It returns the number of seconds elapsed since January 1, 1970.
+```ocaml
+# Unix.time ;;
+- : unit -> float = <fun>
+```
+If you're getting an `Unbound module error` in macOS, run this first: `#require "unix";;`.
+
+To produce its result, no data needs to be passed to that function. The result is entirely determined by external factors. If it was passed information, it would not be used. But something must be passed as a parameter to trigger the request of the current time from the operating system.
+
+Since the function must receive data to trigger the computation but the data is going to be ignored, it makes sense to provide the `unit` value `()`. What is discarded is meaningless in the first place.
+
+A similar reasoning applies to functions producing an effect instead of being externally determined or influenced. Consider `print_endline`. It prints the string it was passed to standard output, followed by a line termination.
+```ocaml
+# print_endline;;
+- : string -> unit = <fun>
+```
+Since the purpose of the function is only to produce an effect, it has no meaningful data to return; therefore, again, it makes sense to return the `unit` value.
+
+This illustrates the relationship between functions intended to have side effects and the `unit` type. The presence of the `unit` type does not indicate the presence of side effects. The absence of the `unit` type does not indicate the absence of side effects. But when no data needs to be passed as input or can be returned as output, the `unit` type should be used to indicate it and suggest the presence of side effects.
+
+## Functions Are Almost as Other Values
+
+Functions are supposed to be exactly as other values; however, there are three restrictions:
+
+1. Function values cannot be displayed in interactive sessions. The placeholder `<fun>` is displayed instead. This is because there is nothing meaningful to print. Once parsed and typed-checked, OCaml discards the function's source code and nothing remains to be printed.
+```ocaml
+# sqrt;;
+- : float -> float = <fun>
+```
+
+2. Equality between functions can't be tested.
+```ocaml
+# pred;;
+- : int -> int = <fun>
+
+# succ;;
+- : int -> int = <fun>
+
```

Comment:
Are `pred` and `succ` known in OCaml like `sqrt` is?
They're a type of _____ that always returns `int -> int = <fun>`?

Reply:
Does that need to be noted? I don't see it in the Tour.

Reply:
I don't think anything is needed here. Typing `pred;;` in utop and having an answer means `pred` was previously defined.